### ISSUE TITLE
Postgres array of single-field UDT reads UDT's field as null instead of actual value

### ISSUE THREAD

**Issue by akervinen** [jOOQ/jOOQ#14010]
### Expected behavior

`jooq.selectFrom(Tables.QUZ).fetch()` in jOOQ 3.13.6:
```
+-----------+
|foos       |
+-----------+
|[row(1.23)]|
+-----------+
```

### Actual behavior

Since 3.14.0: A single-field composite type inside an array has its field read as null.
If the type has more than one field, its fields are read correctly. If the type is not in an array, it's also read correctly.

Running `jooq.selectFrom(Tables.QUZ).fetch()` in jOOQ results in following in 3.17.4:
```
+----------+
|foos      |
+----------+
|[({null})]|
+----------+
```

Raw JDBC returns the value correctly.

### Steps to reproduce the problem

DDL:
```sql
create type foo as
(
    bar real
);

create table quz
(
    foos foo[]
);

insert into quz
values (array [row (1.23)::foo]);
```

Code generation from PostgreSQL instance, default configuration. (Details in `build.gradle.kts` in below repo)

Full repro case: https://github.com/akervinen/jooq-udt-array-repro

### jOOQ Version

jOOQ Open Source 3.17.4

### Database product and version

PostgreSQL 14.5 (Debian 14.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit

### Java Version

openjdk version "17.0.4.1" 2022-08-12

### OS Version

Linux 5.19.9-200.fc36.x86_64

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.5.0

**Comment by lukaseder** [comment]
Thanks a lot for your report. I will look into this, soon

**Comment by lukaseder** [comment]
I can reproduce this. It looks as though our array PG object notation deserialisation doesn't work for some UDTs. In `org.jooq.impl.DefaultBinding.DefaultArrayBinding::pgGetArray`, we rely on an exception being thrown by our `Convert` utility, in case we're trying to read an array of UDTs.

Perhaps, the logic implemented in https://github.com/jOOQ/jOOQ/issues/10229 prevents this exception from being thrown, trying to pass the single value to the single argument constructor of the `UDTRecord`, and failing silently (an often criticised feature of the `Convert` utility).

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0. Will look into backporting the fix, tomorrow. Thanks again for your report!

**Comment by lukaseder** [comment]
Backported to 3.17.5 (https://github.com/jOOQ/jOOQ/issues/14024), and 3.16.11 (https://github.com/jOOQ/jOOQ/issues/14025)
